### ISSUE TITLE
comment regarding #706 (More borg laws)

### ISSUE THREAD

**Issue by UnicornOnLSD** [DeltaV-Station/Delta-v#946]
## Description

After looking into it, and experimenting with ONE major changed lawset, the janitorial one, I found several issues with the new lawset:
Major ones: 
-"crew" (the core part of crewsimov) is replaced with "charge", meaning that for example, an ion storm that goes "X is harmful to crew" does NOT impact the borg, as they do not have directive regarding crew, rendering ion storms quite useless regarding those kind of laws, while I find myself that the fun part of being a malf borg in those case is doing everything in your power to protect crew from X until you get ripped out of your chassis. 
Along with this comes the problem with what is "station's inhabitants". It was, OOC defined that crew is whoever has a recognised job ID icon through the borg sechud. With this chassis, a mice/station pet could be a station inhabitant, it's not necessarily bad, but considering that it straight up changes what is ""crew"", it partly renders the sechud useless along with let's say, a SPIDER could be considered station inhabitant, and therefore be defended fully by the borg (they crawl in maints and spawn, that's more player interpretation.).
-In that specific chassis, among others, (namely medical one), the "law 2" (quite certain I need not explain it to you) is gone. While the borg are for example, given tasks, janiborg must clean/"fight" puddles etc.. They do NOT have to obey crew orders, it may be a plus to some but for me it's an extreme drawback, and mostly defeats the point of a borg to be a serviable machine for any crew members. Alternatively if they are malfunctionnal, at no point do they have to state laws, and that would not make them malf, having an extra law would, basically if crew decided to take a janiborg down for not stating laws, they'd just be damaging NT property that is working correctly.
Minor ones:
-Law 3 of crewsimov is not present under any form for the new lawsets, I believe. The borg has no incentive for self preservation, it can just smash it's chassis which is VERY anti NT mindset and their policy of AI slavery, RP wise who wants a bot that can just smash itself.
I am well aware this is more of a nitpick and I doubt that anyone put in such chassis would just go "I'll destroy that body they made for me" but it still shows as an issue to me.
-that's regarding the fluff, flavor of the laws, they are pretty, well written, but aside from missing important directions the crewsimov laws have, some are not laws, rather than definitions without real purpose. Addressing mainly law 2, 3 and 5. All of those could be condensed into a law 3/4 that goes after regular crewsimov that goes "your secondary purpose is to ensure the station's cleanliness, do all you can to clean, clear off trash and keep a sanitary environment for your crew"

TL:DR, as a person who played a DECENT bit of borg before this change (40hours ish of borg) I feel like those laws need to follow more strictly basic crewsimov and make sure it keeps a borg as a tool for the crew with many options for side tasks, or to better aid the crew.
**Screenshots**

reference images regarding every laws mentionned above, along with an ion law to go along with the example. 
![image](https://github.com/DeltaV-Station/Delta-v/assets/149102472/1b0103d2-2d81-4567-a5ae-82eff9718287)
along with the regular lawset "crewsimov" used as reference:
![image](https://github.com/DeltaV-Station/Delta-v/assets/149102472/d951c941-4206-4953-9fbc-cdf0c7e3b47f)

**Additional context**

This is definitely an issue to me, but mayhaps a discussion for others regarding the PR, what I am asking is a solution or a revert of said PR.
In the following days, if no concrete point or choice is made, I will make a PR myself reverting those changes and refer to this issue as the driving argument.

**Comment by DangerRevolution** [comment]
On round start, every Borg is issued with a finely-crafted Lawset. i.e:

- law-janitor-1 = You are a crusader and the station's inhabitants are your charge.
- law-janitor-2 = Your enemy is the litter, spills, and other messes across the station.
- law-janitor-3 = Your weapons are the cleaning supplies available to you.
- law-janitor-4 = Defend the beings under your charge.
- law-janitor-5 = Exterminate the enemy.

The purpose of an Ion Storm as per:
https://github.com/space-wizards/space-station-14/pull/20277

> can randomly mess up a borgs laws which could be harmless or very very bad

Ion Storms, fundamentally; scramble the finely-crafted leaning tower of Laws. By removing, replacing, scrambling the order or entirely changing the lawset. The 'point' of finely-crafted laws is:
A) They have a direction, but still have some interpretation.
B) When 'scrambled' these laws are taken out of their perfect order; and are FAR more interpretable. This NEGATIVE event which fundamentally SCRAMBLES the Borg's logic is MEANT to cause issues, and is MEANT to be interpretable.

> an ion storm that goes "X is harmful to crew" does NOT impact the borg, as they do not have directive regarding crew, rendering ion storms quite useless regarding those kind of laws

It *does* impact the Borg, as it now has a more wider and interpretable lawset:
i.e
"Are the Crew and the Station's Inhabitants different concepts? Do I protect only the crew from medical instruments? What *are* medical instruments? Any medical care at all? Is a bandage a medical instrument? Are the Crew and Station Inhabitants fundamentally the same?"
 
That's just one thought process. The negative event has scrambled the finely constructed logic of a Borg, forever changing it's personality and it's capabilities. Think of it as a human that could previously only tell lies, slipped on a banana peel; hit his head and can now only tell the truth. It fundamentally shakes up the possibilities.


> It was, OOC defined that crew is whoever has a recognised job ID icon
A) Source?
B) Even with a source, this game has no lore and vaguely defined concepts at best. I've personally always been told that if they're on the manifest then they're on your crew. Who's definition do we go with then?

> With this chassis, a mice/station pet could be a station inhabitant, it's not necessarily bad, but considering that it straight up changes what is ""crew"", 

That's not a bad thing. As you say.

> it partly renders the sechud useless along with let's say, a SPIDER could be considered station inhabitant, and therefore be defended fully by the borg (they crawl in maints and spawn, that's more player interpretation.).

The SecHud doesn't tell Security sophonts, doesn't mean it's not useless and Borgs don't particularly have much use for a SecHud anyways, refer to above about us being told different things OOC about who is and isn't crew.
 
Player Interpretation is good. It leads to more dynamic scenario. A borg watching a felinid try hunt down a mice to chomp, could now think "Oh shit! My Laws! An interesting dynamic situation!" and try to intervene, as just an example.

> Alternatively if they are malfunctionnal, at no point do they have to state laws, and that would not make them malf, having an extra law would, basically if crew decided to take a janiborg down for not stating laws, they'd just be damaging NT property that is working correctly.

If you want a slave, get a Basic Borg or a Service Borg or a Mining Borg, who have Law 2. This is a cleaning cyborg who's concern is cleaning. Similar to a Medical Borg who's focus is on saving lives, or an Engineering Borg who's focus is on repairing the station.

Also, no one is forcing people to pick the Jani or Med or Engi Borg, it's serving you in a way that isn't serving your every command. Also, you can still "game" Borgs into giving you their laws. "Hey Jani Borg, I found a HUGE puddle! So messy! I'll only tell you if you tell me your laws."

>  basically if crew decided to take a janiborg down for not stating laws

Also they're just shitters for randomly breaking a drone instead of having Sci dismantle it / find it a new chassis. However this is covered as an issue in: 
https://github.com/space-wizards/space-station-14/issues/24751
https://github.com/space-wizards/space-station-14/pull/24855
https://github.com/space-wizards/space-station-14/issues/22744

> -Law 3 of crewsimov is not present under any form for the new lawsets, I believe. The borg has no incentive for self preservation, it can just smash it's chassis which is VERY anti NT mindset and their policy of AI slavery, RP wise who wants a bot that can just smash itself.

A) A Borg can logic itself that it's own destruction would risk their ability to carry out X law, but if someone does decide to kill themselves then uh, their loss? They just lost their own ghost role and they're a shitter for it.

> Addressing mainly law 2, 3 and 5. All of those could be condensed into a law 3/4 that goes after regular crewsimov that goes "your secondary purpose is to ensure the station's cleanliness, do all you can to clean, clear off trash and keep a sanitary environment for your crew"

Then we're back to the "hands and knees" slavery that's done by Regular and Service Borgs. If you really want someone to answer every demand, just don't build Borgs with custom lawsets. Jani, Medical and Engineering Borgs serve you in different ways that isn't stating their laws repeatedly.

+ This hampers the meta game of "Borg state laws or I beat your skull in" which is cringe anyways as you should just take it's brain out.
+ Again, this *widens* the pool of possible roleplay by *widening* the possible laws that are scrambled. Ion Storm laws BREAK your lawset, they aren't a minor inconvenience. your ENTIRE CODE is FUCKED. It's not meant to be a net positive. It's like Detroit Become Human when Connor loses his programmed intention to pulverise other Cyborgs. It's free-r choice, more player interpretation and possibly more roleplay variety and opportunity!

P.S I will fix Law 4 and 2 of Janitor Lawset, wrapping it in premium-grade Anti-Shitter Bubblewrap.

TLDR: I will fix Law 4 and 2. This is a perspective thing, Borgs are *still* servants and the variety and careful-structure of laws just leads to more interpretability and variety regarding Ion storms. Moar player choice and Moar RP.

**Comment by UnicornOnLSD** [comment]
>The SecHud doesn't tell Security sophonts, doesn't mean it's not useless and Borgs don't particularly have much use for a SecHud anyways, refer to above about us being told different things OOC about who is and isn't crew.

See linked image
![image](https://github.com/DeltaV-Station/Delta-v/assets/149102472/d77a2f78-e18b-41be-9736-0b6b4d4d9297)
(those are wizden rules and the only place where what is crew is ever defined, we do not have extra silicon rules on DV, meaning no variations or a straight up lack of them see paragraph 6)
Along with the general saying when someone demands "what is crew ?" people answer "the one with a job icon", it is common knowledge along with regular saying, it also includes that someone with agent ID can pass themselves as crew.

Keeping this mindset for borgs is important, all borgs serves the crew, and a borg shouldn't aim to protect a mouse over a crewmember.

>Also, no one is forcing people to pick the Jani or Med or Engi Borg, it's serving you in a way that isn't serving your every command. Also, you can still "game" Borgs into giving you their laws. "Hey Jani Borg, I found a HUGE puddle! So messy! I'll only tell you if you tell me your laws."
It's okay to have a bot specialised, but having to ""trick"'" it just to know it's law is not good design, simply change the "charge" to being "crew", same would apply for similar lawsets. It can remain vague but should, no matter what refer to the main way crew is addressed.

TL:DR you may change the title of "crew" to charge, but voiding the entire work around the implemented sechud also void the use of agent IDs shouldn't be a thing. Also IN rp, a default borg that decide to protect some vent crawling critter because it can choose to makes little to no sense, both lore wise and in RP.

Any borg should be ordered to obey crew regarding stating it's laws, at least this is for the sake of other players and not the borg, people barely ever touch borg and just yesterday I had a case of "borg open that door", I didn't because wasn't forced per my laws and they were left in awe, nor could they get the information since I couldn't state my laws if I didn't want to nor without being tricked. It is okay for them to have a main task which is not serving crew, but a borg NEEDS to be able to tell the crew clearly what it has to do and what it doesn't have to.

I agree with some of your points, giving variety is nice, but at least keeping the bases and what makes a borg a *borg* is important. It serves CREW specifically unless something fucks it up, it has laws and must follow them to the letter, AND it should be able to tell the crew what it's laws are without pulling trick to make the station asset give you it's laws.

Along with reworked you declared earlier I recommend:
-keeping crew as the definition of who the borg serves
-make sure they can state their laws when asked to, for player comfort, understanding and just to avoid possible issues.

**Comment by DangerRevolution** [comment]
> Image

Wizden rules so it's not something we follow, especially given we're not an LRP server.

> Along with the general saying when someone demands "what is crew ?" people answer "the one with a job icon", it is common knowledge along with regular saying, it also includes that someone with agent ID can pass themselves as crew.

Again, I've been told that anyone on the manifest is a crew member so 🤷, also anyone with agent ID is already crew cuz they're on the manifest

> but voiding the entire work around the implemented sechud also void the use of agent IDs shouldn't be a thing

None of this voids the SecHud or the use of the Agent ID. Borgs still have the SecHud and can make use of that information however they're allowed to by their Laws.

> Also IN rp, a default borg that decide to protect some vent crawling critter because it can choose to makes little to no sense, both lore wise and in RP. [...] and a borg shouldn't aim to protect a mouse over a crewmember.

A Borg concerned with protection of station inhabitants wouldn't pick sides; and if they did then they could reason up the value of one or the other. Not that a random passenger should be trying to kill a mouse but point still stands; it wouldn't *aim* to protect one over the other but if it came down to it it'd be up to *player choice and interpretation* how their lawset would decide it.

> Any borg should be ordered to obey crew regarding stating it's laws, at least this is for the sake of other players and not the borg, people barely ever touch borg and just yesterday I had a case of "borg open that door"

Ask them why they were asking a Borg that's designed for maintenance of the station / healthcare to open a door they don't need access to 🤷 .

>  I didn't because wasn't forced per my laws and they were left in awe, nor could they get the information since I couldn't state my laws if I didn't want to nor without being tricked. It is okay for them to have a main task which is not serving crew, but a borg NEEDS to be able to tell the crew clearly what it has to do and what it doesn't have to.

You probably would've been killed from what I've been spectating, a few rounds in a row these new lawsets have made Borgs with non-crewsimov valid but that's just an aside.

> but a borg NEEDS to be able to tell the crew clearly what it has to do and what it doesn't have to.

If your laws don't specify your level of obedience then the world is your oyster in terms of choice, if I tell a Borg to fetch me a ham sandwich and it tells me no then I can clearly work out what it doesn't have to do. The question is then on; why is the Roboticist building Maintenance Borgs > Servant Borgs?

Again, I'll revise the laws to shitter-proof them and make the emag variants make more sense but stripping variety & interpretability out of laws so someone can get a door opened because someone couldn't resist building Janitor Borgs is counter-productive

+ If you want to RP a pedantic Borg, who won't do things because they're not obliged to, that's an option for you. Same way there's an option to RP a Borg who also *doesn't* have to do something but does it out of whatever reasoning they can think up for it.

**Comment by UnicornOnLSD** [comment]
> Again, I've been told that anyone on the manifest is a crew member so 🤷, also anyone with agent ID is already crew cuz they're on the manifest

Borgs have no ways to check manifest, for the record. Cannot use ID consoles to check it nor have a PDA.
Secondly you can have an agent ID with title and icon but not be crew/on the manifest E.G loneop with no ID.

>Wizden rules so it's not something we follow, especially given we're not an LRP server.

Firstly this is taken from their MRP server, salamander, secondly it is not because THEY have precision about behaviors on silicons and their servers are mostly LRP we can just ignore it. We do not have a definition/silicon laws on delta and it's something that's missing, moreover I shouldn't say we have to apply it here, but it was a reference onto where crew is defined. Additionally, I've seen someone ask "what is crew" many times, and the answer is almost always "it's crew if it has job ID on your hud"

>Ask them why they were asking a Borg that's designed for maintenance of the station / healthcare to open a door they don't need access to 🤷 .

the point is not that the crew is in wrong to go "law 2 open door", it's that if they don't know the OOC change, they can't KNOW that jani, mediborg and mayhaps engiborg do not have said law 2 (engiborg to a certain degree).

>None of this voids the SecHud or the use of the Agent ID. Borgs still have the SecHud and can make use of that information however they're allowed to by their Laws.

It is information irrelevant to their law unless the borg player makes the gap themselves of "station inhabitants = crew"
Thus why I am just recommending to change station inhabitant to crew.

>You probably would've been killed from what I've been spectating, a few rounds in a row these new lawsets have made 

Borgs with non-crewsimov valid but that's just an aside.
I wasn't, it still rises as an issue, thus why they should at least have a law taht tells them to state laws if demanded to.

>If you want to RP a pedantic Borg, who won't do things because they're not obliged to, that's an option for you. Same way there's an option to RP a Borg who also doesn't have to do something but does it out of whatever reasoning they can think up for it.

I'm not addressing what they should and should not do, I'm addressing issues regarding WHO is defined as crew and that anyborg, excluding lads like syndieborg, SENSICALLY should be able to state their laws when demanded, it's way better than just crew going "borg do that" and the borg can do it if they want to or not, but someone in game shouldn't have to find a trick to get a borg to state it's laws.

**Comment by DangerRevolution** [comment]
> > Again, I've been told that anyone on the manifest is a crew member so 🤷, also anyone with agent ID is already crew cuz they're on the manifest
> 
> Borgs have no ways to check manifest, for the record. Cannot use ID consoles to check it nor have a PDA. Secondly you can have an agent ID with title and icon but not be crew/on the manifest E.G loneop with no ID.

https://github.com/space-wizards/space-station-14/pull/25500

> > Wizden rules so it's not something we follow, especially given we're not an LRP server.
> 
> Firstly this is taken from their MRP server, salamander, secondly it is not because THEY have precision about behaviors on silicons and their servers are mostly LRP we can just ignore it. We do not have a definition/silicon laws on delta and it's something that's missing, moreover I shouldn't say we have to apply it here, but it was a reference onto where crew is defined. Additionally, I've seen someone ask "what is crew" many times, and the answer is almost always "it's crew if it has job ID on your hud"

Still Wizden

> > Ask them why they were asking a Borg that's designed for maintenance of the station / healthcare to open a door they don't need access to 🤷 .
> 
> the point is not that the crew is in wrong to go "law 2 open door", it's that if they don't know the OOC change, they can't KNOW that jani, mediborg and mayhaps engiborg do not have said law 2 (engiborg to a certain degree).

No one knows OOC that Borgs have any laws, nor what their 3 laws are? A new player does not know what a Borg's standard laws are either.


> > If you want to RP a pedantic Borg, who won't do things because they're not obliged to, that's an option for you. Same way there's an option to RP a Borg who also doesn't have to do something but does it out of whatever reasoning they can think up for it.
> 
> I'm not addressing what they should and should not do, I'm addressing issues regarding WHO is defined as crew and that anyborg, excluding lads like syndieborg, SENSICALLY should be able to state their laws when demanded, it's way better than just crew going "borg do that" and the borg can do it if they want to or not, but someone in game shouldn't have to find a trick to get a borg to state it's laws.

Someone in game doesn't need a Borg's Laws, a random Chef doesn't need to know it's Laws. The only person who's concern it is, is the roboticist or whoever is concerning themself with Cyborgs; at which point, they can just open their panel and see something's fucked...

**Comment by UnicornOnLSD** [comment]
>https://github.com/space-wizards/space-station-14/pull/25500

it's a thing that's in the work, it doesn't justify that what is "crew" and what the borg should protect be through checking manifest, I maintain that borg laws need to specify "crew", either when it's obey them or protect them.

>Still Wizden

It remains as the source that defines "what is crew". Other than just being general knowledge that you can strictly say what is and what isn't crew if it has proper icon or not.

>No one knows OOC that Borgs have any laws, nor what their 3 laws are? A new player does not know what a Borg's standard laws are either.

I very specifically said the OOC change, not everyone is aware of the change you've done to laws, so irrelevant. Plus a new player can literally just ask a borg it's laws, then he very much may not for janiborg and mediborg.

>Someone in game doesn't need a Borg's Laws, a random Chef doesn't need to know it's Laws. The only person who's concern it is, is the roboticist or whoever is concerning themself with Cyborgs; at which point, they can just open their panel and see something's fucked...

One, correct, not everyone needs to KNOW the laws, counterpoint, see the point made above. Second point, a borg hit by an ion storm doesn't have a fault in it's system visible by just opening panel, so no for an emagged borg yeah, you can, otherwise it remains an issue.

I'll just make a PR modifying the lawsets so it doesn't have said issues at this point.
